sormas-backend/src/main/java/de/symeda/sormas/backend/docgeneration/DocumentTemplateEntitiesBuilder.java

```diff
@@ -0,0 +1,233 @@
+		}
+	}
+
+	public Map<ReferenceDto, DocumentTemplateEntities> getQuarantineOrderEntities(DocumentWorkflow workflow, List<ReferenceDto> referenceDtos)
+		throws DocumentTemplateException {
+		List<String> entityUuids = referenceDtos.stream().map(ReferenceDto::getUuid).collect(Collectors.toList());
+
+		final Stream<AbstractMap.SimpleEntry<ReferenceDto, DocumentTemplateEntities>> entities;
+
+		final BulkEntitiesBuilder<? extends EntityDto> builder;
+		switch (workflow) {
+		case QUARANTINE_ORDER_CASE:
+			builder = createBulkEntitiesBuilder(
+				ROOT_CASE,
+				caseFacade.getByUuids(entityUuids),
+				CaseDataDto::toReference,
+				CaseDataDto::getPerson,
+				new SampleCriteria().caseUuids(entityUuids));
+
+			break;
+		case QUARANTINE_ORDER_CONTACT:
+			builder = createBulkEntitiesBuilder(
+				ROOT_CONTACT,
+				contactFacade.getByUuids(entityUuids),
+				ContactDto::toReference,
+				ContactDto::getPerson,
+				new SampleCriteria().contactUuids(entityUuids));
+
+			break;
+		case QUARANTINE_ORDER_EVENT_PARTICIPANT:
+			builder = createBulkEntitiesBuilder(
+				ROOT_EVENT_PARTICIPANT,
+				eventParticipantFacade.getByUuids(entityUuids),
+				EventParticipantDto::toReference,
+				(EventParticipantDto e) -> e.getPerson().toReference(),
+				new SampleCriteria().eventParticipantUuids(entityUuids));
+
+			break;
+		default:
+			throw new DocumentTemplateException(I18nProperties.getString(Strings.errorQuarantineOnlyCaseAndContacts));
+		}
+
+		return builder.build();
+	}
+
+	<T extends EntityDto> BulkEntitiesBuilder<T> createBulkEntitiesBuilder(
+		RootEntityType mainRootEntityType,
+		List<T> mainRootEntities,
+		Function<T, ReferenceDto> createReference,
+		Function<T, PersonReferenceDto> getEntityPerson,
+		SampleCriteria sampleCriteria) {
+
+		return new BulkEntitiesBuilder<>(mainRootEntityType, mainRootEntities, createReference, getEntityPerson, sampleCriteria);
+	}
+
+	final class BulkEntitiesBuilder<T extends EntityDto> {
+
+		final RootEntityType mainRootEntityType;
+		final List<T> mainRootEntities;
+		final Function<T, ReferenceDto> createReference;
+		final Function<T, PersonReferenceDto> getEntityPerson;
+		final SampleCriteria sampleCriteria;
+
+		private BulkEntitiesBuilder(
+			RootEntityType mainRootEntityType,
+			List<T> mainRootEntities,
+			Function<T, ReferenceDto> createReference,
+			Function<T, PersonReferenceDto> getEntityPerson,
+			SampleCriteria sampleCriteria) {
+			this.mainRootEntityType = mainRootEntityType;
+			this.mainRootEntities = mainRootEntities;
+			this.createReference = createReference;
+			this.getEntityPerson = getEntityPerson;
+			this.sampleCriteria = sampleCriteria;
+		}
+
+		Map<ReferenceDto, DocumentTemplateEntities> build() {
+			Map<String, SampleDto> samples =
+				sampleFacadeEjb.getPositiveOrLatest(sampleCriteria).stream().collect(Collectors.toMap(s -> s.getAssociatedCase().getUuid(), s -> s));
```

Comment:
A sample will not always be associated with a case directly so this might fail with NPE